build_modules/lib/src/modules.dart

```diff
@@ -25,6 +26,26 @@ part 'modules.g.dart';
 @_AssetIdConverter()
 @_DartPlatformConverter()
 class Module {
+  /// Merge the sources and dependencies from [modules] into a single module.
+  ///
+  /// The resulting modules will have a [primarySource] which is the earliest
+  /// value from [sources] if it were sorted. It will have the same [isMissing],
+  /// [isSupported] and [platform] as the first entry. The [sources] and
+  /// [directDependencies] will be merged, and dependencies will be filtered so
+  /// that no source is a dependency.
+  static Module merge(List<Module> modules) {
+    assert(modules.isNotEmpty);
+    if (modules.length == 1) return modules.single;
+    final allSources = Set.of(modules.expand((m) => m.sources));
+    final allDependencies = Set.of(modules.expand((m) => m.directDependencies))
+      ..removeAll(allSources);
+    final primarySource =
+        allSources.reduce((a, b) => a.compareTo(b) < 0 ? a : b);
+    return Module(primarySource, allSources, allDependencies,
+        modules.first.platform, modules.first.isSupported,
```

Comment:
This was broken before but `isSupported` should actually be `modules.every((m) => m.isSupported)`